.pre-commit-config.yaml

```diff
@@ -0,0 +1,17 @@
+files: ^client/python/
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-ast
+      - id: check-builtin-literals
+      - id: check-merge-conflict
+      - id: check-toml
+      - id: debug-statements
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: "v0.0.265"
+    hooks:
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
```

Comment:
How about we add `insert-license` from `repo: https://github.com/Lucas-C/pre-commit-hooks`? We can do that in separate PR as well, just seems a nice addition while introducing pre-commit at all.

Reply:
Would do it as follow-up as is orthogonal  👍